### PR TITLE
CAS-11711 C++ upgrades/reformat/refactor no functionality changes

### PR DESCRIPTION
**Pull request by dmehring**

### CODE REVIEW

**Comment by dpetry** [comment]
Tammo, if you are not on vacation, can you check what the check failure is about?

**Comment by dpetry** [comment]
Hi Dave, while we are waiting for Ger or Tammo to tell us what the Travis failure means,
I started looking at the code you submitted. These are very comprehensive changes.
Could you briefly tell me what the philosophy was here and how you convinced yourself that the new code is better?

**Comment by dmehring** [comment]
Hi Dirk
The initial motivation was to perform C++11 upgrades now that casacore supports C++11. While I was in the code, I also reformatted it so lines do not span more than 80 characters, at least for the most part. I added some aliases to help shorten often repeated types. These changes are akin to changes Martin approved a couple weeks ago for other files in this module. So it’s just a continuation of that work. There are no feature or behavior changes.

**Comment by dpetry** [comment]
Did you do any tests besides running the existing unit tests?

**Comment by dmehring** [comment]
No. The existing unit tests are comprehensive.

**Comment by dpetry** [comment]
Maybe, while we are waiting for Travis, you could run one or two selected CASA regression tests which make use of this library?

File: scimath/StatsFramework/StatisticsDataset.h [hunk 1]
```diff
@@ -255,32 +254,32 @@ class StatisticsDataset {
...
+    std::map<uInt, Bool> _isIncludeRanges{};
+    std::map<uInt, DataRanges> _dataRanges{};
+    StatsDataProvider<CASA_STATP>* _dataProvider{nullptr};
+
+    Int64 _idataset{0};
+    typename std::vector<DataIterator>::const_iterator _dend{}, _diter{};
+    std::vector<Int64>::const_iterator _citer{};
+    std::vector<uInt>::const_iterator _dsiter{};
+    uInt _dataCount{0};
+    ChunkData _chunk{};
```
Comment (tammojan) [inline scimath/StatsFramework/StatisticsDataset.h]: This change breaks the clang Travis build, see https://travis-ci.org/casacore/casacore/jobs/414161122#L8291

**Comment by tammojan** [comment]
In case following the Travis links doesn't work for you, here is the relevant clang error:
```
[ 57%] Building CXX object scimath/StatsFramework/test/CMakeFiles/tClassicalStatistics.dir/tClassicalStatistics.cc.o
In file included from /home/travis/build/casacore/casacore/scimath/StatsFramework/test/tFitToHalfStatistics.cc:28:
In file included from /home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/FitToHalfStatistics.h:31:
In file included from /home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/ConstrainedRangeStatistics.h:31:
In file included from /home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/ClassicalStatistics.h:31:
In file included from /home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/StatisticsAlgorithm.h:34:
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/StatisticsDataset.h:282:22: error: chosen constructor is explicit in copy-initialization
    ChunkData _chunk{};
                     ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/StatisticsDataset.h:77:5: note: in instantiation of default member initializer 'casacore::StatisticsDataset<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::_chunk' requested here
    StatisticsDataset();
    ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/StatisticsAlgorithm.h:390:43: note: in instantiation of member function 'casacore::StatisticsDataset<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::StatisticsDataset' requested here
    StatisticsDataset<CASA_STATP> _dataset{};
                                          ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/ClassicalStatistics.tcc:56:5: note: in instantiation of member function 'casacore::StatisticsAlgorithm<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::StatisticsAlgorithm' requested here
) : StatisticsAlgorithm<CASA_STATP>(),
    ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/ConstrainedRangeStatistics.tcc:38:5: note: in instantiation of member function 'casacore::ClassicalStatistics<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::ClassicalStatistics' requested here
) : ClassicalStatistics<CASA_STATP>(qc) {
    ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/FitToHalfStatistics.tcc:46:5: note: in instantiation of member function 'casacore::ConstrainedRangeStatistics<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::ConstrainedRangeStatistics' requested here
) : ConstrainedRangeStatistics<CASA_STATP>(
    ^
/home/travis/build/casacore/casacore/scimath/StatsFramework/test/tFitToHalfStatistics.cc:1914:101: note: in instantiation of member function 'casacore::FitToHalfStatistics<double, casacore::Array<double>::ConstIteratorSTL, casacore::Array<bool>::ConstIteratorSTL, casacore::Array<double>::ConstIteratorSTL>::FitToHalfStatistics' requested here
            FitToHalfStatistics<Double, Array<Double>::const_iterator, Array<Bool>::const_iterator> fh(
                                                                                                    ^
/home/travis/build/casacore/casacore/casacore/casa/Arrays/Array.h:799:16: note: explicit constructor declared here
      explicit ConstIteratorSTL (const T* end = 0)
               ^
/home/travis/build/casacore/casacore/casacore/scimath/StatsFramework/StatisticsDataset.h:62:22: note: in implicit initialization of field 'data' with omitted initializer
        DataIterator data;
                     ^
1 error generated.
```

**Comment by dmehring** [comment]
Thanks Tammo I'll have a look.
Dirk, I've run various CASA tests for this, in particular ones which are very intensive with respect to the exercise of this code. They pass. I'm busy with 5.4 issues otherwise atm. All tests will be run prior to it being merged into CASA. If there are any casacore related failures, I'll have to fix them here before proceeding.
Thanks

**Comment by dpetry** [comment]
OK, that should be enough testing.

**Comment by dmehring** [comment]
Thanks Dirk.

**Pull request merged**